scripts/mvftoms.py

```diff
@@ -241,47 +271,48 @@ def _cp_index(a1, a2, pol):
+    applycal_products = ', '.join(getattr(dataset, '_applycal_products', []))
+    if applycal_products:
+        print('The following calibration products will be applied:', applycal_products)
+    else:
+        print('No calibration products will be applied')
 
-    # Get list of unique polarisation products in the file
-    pols_in_file = np.unique([(cp[0][-1] + cp[1][-1]).upper() for cp in dataset.corr_products])
+    # Get list of unique polarisation products in the dataset
+    pols_in_dataset = np.unique([(cp[0][-1] + cp[1][-1]).upper() for cp in dataset.corr_products])
 
     # Which polarisation do we want to write into the MS
     # select all possible pols if full-pol selected, otherwise the selected polarisations via pols_to_use
     # otherwise finally select any of HH,VV present (the default).
     pols_to_use = ['HH', 'HV', 'VH', 'VV'] if (options.full_pol or options.circular) else \
         list(np.unique(options.pols_to_use.split(','))) if options.pols_to_use else \
-        [pol for pol in ['HH', 'VV'] if pol in pols_in_file]
+        [pol for pol in ['HH', 'VV'] if pol in pols_in_dataset]
 
     # Check we have the chosen polarisations
-    if np.any([pol not in pols_in_file for pol in pols_to_use]):
+    if np.any([pol not in pols_in_dataset for pol in pols_to_use]):
         raise RuntimeError("Selected polarisation(s): %s not available. "
                            "Available polarisation(s): %s"
-                           % (','.join(pols_to_use), ','.join(pols_in_file)))
+                           % (','.join(pols_to_use), ','.join(pols_in_dataset)))
 
     # Set full_pol if this is selected via options.pols_to_use
     if set(pols_to_use) == set(['HH', 'HV', 'VH', 'VV']) and not options.circular:
         options.full_pol = True
 
-    pol_for_name = 'full_pol' if options.full_pol else \
-                   'circular_pol' if options.circular else \
-                   '_'.join(pols_to_use).lower()
-
+    # Extract one MS per spectral window in the dataset(s)
     for win in range(len(dataset.spectral_windows)):
         dataset.select(reset='T')
 
-        # Extract MS file per spectral window in observation file
         freq_MHz = dataset.spectral_windows[win].centre_freq / 1e6
         print('Extract MS for spw %d: central frequency %.2f MHz' % (win, freq_MHz))
 
-        # If no output MS filename supplied, infer the output filename
-        # from the first dataset.
+        # If no output MS directory name supplied, infer it from dataset(s)
         if options.output_ms is None:
-            # create MS in current working directory
-            ms_name = '%s_%d.%s%s.ms' % (
-                os.path.splitext(args[0])[0], freq_MHz,
-                "" if len(args) == 1 else "et_al.", pol_for_name)
+            if len(dataset.spectral_windows) > 1:
+                # Use frequency label to disambiguate multiple spectral windows
+                ms_name = default_ms_name(args, freq_MHz)
```

Comment:
What if there are two spectral windows within 1MHz of each other?

Reply:
Sure. This is a unlikely event (windows within 1 MHz) on top of an unlikely event (a dataset even having more than one spectral window). I'll add a few digits then. Kilohertz should be safe :-)